aqueduct/gateway/tests/test_endpoints.py

```diff
@@ -1154,3 +1157,722 @@ def test_user_rate_limit_output_tokens_per_minute(self):
+        mock_client = MagicMock()
+
+        mock_transcription = Transcription(text="Mock transcription text")
+
+        # Create an async mock for the transcription creation
+        async def mock_create(**kwargs):
+            return mock_transcription
+
+        mock_client.audio.transcriptions.create = mock_create
+
+        with patch("gateway.config.get_router_config", return_value=mock_config):
+            with patch("gateway.views.transcriptions.get_router", return_value=mock_router):
+                with patch(
+                    "gateway.views.transcriptions.get_openai_client", return_value=mock_client
+                ):
+                    with patch(
+                        "litellm.get_llm_provider", return_value=(self.stt_model, "openai", "", "")
+                    ):
+                        headers = _build_chat_headers(self.AQUEDUCT_ACCESS_TOKEN)
+                        # Remove Content-Type for multipart form data
+                        headers.pop("Content-Type", None)
+
+                        response = self.client.post(
+                            "/audio/transcriptions",
+                            {
+                                "file": test_audio_file,
+                                "model": "stt",
+                            },  # Using alias instead of actual model name
+                            headers=headers,
+                        )
+
+                        # Should return 200 with alias resolution
+                        self.assertEqual(
+                            response.status_code,
+                            200,
+                            f"Alias resolution not working for STT. Expected 200, got {response.status_code}: {response.content}",
+                        )
+
+                        response_json = response.json()
+                        self.assertIn("text", response_json, "Response should contain 'text' field")
+
+                        # Check that the database contains one request
+                        requests = list(Request.objects.all())
+                        self.assertEqual(
+                            len(requests),
+                            1,
+                            "There should be exactly one request after transcription.",
+                        )
+
+    def test_chat_completion_with_nonexistent_alias(self):
+        """
+        Test that requests with non-existent aliases return appropriate errors.
+        """
+        messages = [{"role": "user", "content": "Hello!"}]
+
+        headers = _build_chat_headers(self.AQUEDUCT_ACCESS_TOKEN)
+        payload = {"model": "nonexistent-alias-12345", "messages": messages, "max_tokens": 10}
+
+        response = self.client.post(
+            "/chat/completions",
+            data=json.dumps(payload),
+            headers=headers,
+            content_type="application/json",
+        )
+
+        # Should return 400 (Bad Request) or 404 (Model not found)
```

Comment:
What does the response status code depend on? Isn't it going to be always the same?